hack/generator/pkg/codegen/pipeline/flatten_properties.go

```diff
@@ -52,7 +53,8 @@ func makeFlatteningVisitor(defs astmodel.Types) astmodel.TypeVisitor {
 
 			// safety check:
 			if err := checkForDuplicateNames(newProps); err != nil {
-				return nil, err
+				klog.V(1).Infof("Flattening caused duplicate property names, skipping flattening")
```

Comment:
Maybe this is an error or warning rather than info?

Also do we want to file a bug to do something more intelligent here in the future?

Reply:
Yes I'm going to open a task.